beets/importer.py

```diff
@@ -1225,7 +1225,33 @@ def handlers(cls):
             cls._handlers.append((is_zipfile, ZipFile))
             import tarfile
 
-            cls._handlers.append((tarfile.is_tarfile, tarfile.open))
+            class TarInfo(tarfile.TarInfo):
+                def __init__(self, member):
+                    super().__init__(member.name)
+                    self.name = member.name
+                    self.size = member.size
+                    self.mode = member.mode
+                    self.mtime = member.mtime
+                    self.type = member.type
+                    self.linkname = member.linkname
+                    self.uid = member.uid
+                    self.gid = member.gid
+                    self.uname = member.uname
+                    self.gname = member.gname
+                    self.date_time = time.gmtime(member.mtime)[:6]
```

Comment:
I think this could be stripped down to just the bare minimum that we need here: Thus, ` TarInfo` can be a new class (not wrap the original `TarInfo`), and only set `date_time` and `filename`.